GoogleCloudPlatform/microservices-demo
# K8s

### Background
Before this pull request, the microservices-demo project did not include support for deploying microservices using Helm charts and managing Redis using Helm charts. Additionally, the project did not have readiness and liveness probes configured for the microservices.

### Fixes
This pull request addresses the following issues:
- Adds Helm charts for deploying the microservices and Redis.
- Configures services and values for the Helm charts.
- Adds a Helmfile to manage the deployment of Helm charts.
- Configures readiness and liveness probes for the microservices.

### Change Summary
- Added Helm charts for the microservices and Redis.
- Configured services and values for the Helm charts to customize the deployments.
- Created a Helmfile to manage the deployment of Helm charts.
- Implemented readiness and liveness probes for the microservices to ensure their health and availability.

### Additional Notes
The Helm charts provide an efficient way to deploy and manage the microservices and Redis in a Kubernetes cluster. By utilizing Helm and Helmfile, the deployment process becomes more streamlined and manageable.

### Testing Procedure
To test the changes:
1. Ensure that the necessary prerequisites are in place, such as a running Kubernetes cluster and Helm installed.
2. Deploy the microservices and Redis using the Helmfile command (`helmfile sync`) and verify that all the resources are created successfully.
3. Validate that the microservices and Redis are running as expected within the cluster.
4. Test the functionality of the microservices and Redis to ensure that they are working correctly.
5. Verify the readiness and liveness probes by checking the health status of the microservices and ensuring they respond appropriately.

### Related PRs or Issues
None.
